### PR TITLE
[EAGLE-5016]: New dockerfile

### PR DESCRIPTION
**Pull request by zeiler**
This updates dockerfiles to use debian and simplifies them. 

Added a flag to model upload to skip generating the dockerfile from template. This is helpful when debugging docker build or runtime issues as you can set that flag after manually editting the generated dockerfile. 

Also skip packaging up checkpoints unless --download_checkpoints is on during upload so that re-using folders for multiple uploads is easier.

### CODE REVIEW

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (wemoveon2) [inline clarifai/runners/dockerfile_template/Dockerfile.cuda.template:22]: fixed ver?

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (wemoveon2) [inline clarifai/runners/dockerfile_template/Dockerfile.cuda.template:8]: should use `TARGETPLATFORM` which gets set by docker instead

`ARG` also need to be after `FROM`

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (wemoveon2) [inline clarifai/runners/dockerfile_template/Dockerfile.cuda.template:53]: should use `*-linux-gnu`

Comment (wemoveon2) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:53]: *where globbing is possible like for `COPY`

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (wemoveon2) [inline clarifai/runners/dockerfile_template/Dockerfile.cuda.template:1]: missing all the symlink libs

Comment (wemoveon2) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:1]:
```docker
FROM gcr.io/distroless/python3-debian12
ARG PYTHON_VERSION
# files for user profiles
COPY --from=build /etc/passwd /etc/passwd
COPY --from=build /etc/group /etc/group
# needed to call pip directly
COPY --from=build --chown=appuser:appuser /bin/bash /bin/sh
# virtual env
COPY --from=build --chown=appuser:appuser /venv /venv
# we have to overwrite the python3 binary that the distroless image uses
COPY --from=build /usr/bin/python${PYTHON_VERSION} /usr/bin/python3
# also copy in all the lib files for it.
COPY --from=build /usr/lib/python${PYTHON_VERSION} /usr/lib/python${PYTHON_VERSION}
# ld.so is needed for anything using shared libraries, automounted CUDA packages won't work without it
COPY --from=build /lib/*-linux-gnu/ld-linux-*.so.2 /lib/
COPY --from=build /lib/*-linux-gnu/libc.so.6 /lib/
COPY --from=build /lib/*-linux-gnu/libdl.so.2 /lib/
COPY --from=build /lib/*-linux-gnu/libpthread.so.0 /lib/
COPY --from=build /lib/*-linux-gnu/librt.so.1 /lib/
# ldconfig is needed to update the shared library cache so system libraries (like CUDA) can be found
COPY --from=build /usr/sbin/ldconfig /sbin/ldconfig
COPY --from=build /etc/ld.so.conf /etc/ld.so.conf
COPY --from=build /etc/ld.so.cache /etc/ld.so.cache
# ln is needed to create symbolic links (needed by nvidia-container-runtime)
COPY --from=build /usr/bin/ln /usr/bin/ln
# Set environment variables to use virtualenv by default
ENV VIRTUAL_ENV=/venv
ENV PYTHONPATH=${PYTHONPATH}:${VIRTUAL_ENV}/lib/python3.11/site-packages
ENV PATH=${VIRTUAL_ENV}/bin:/bin/bash:/bin/sh:${PATH}
# Switch to the non-root user
USER appuser
SHELL ["/bin/bash", "-c"]
RUN pip install torch
```

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (zeiler) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:8]: it seems to work when it's before. but maybe it's because we're templating it it just "works"?

Comment (zeiler) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:8]: seems like ARG can be before FROM in their docs. I think we need the target platform to be set indendent of where the build is happening so we can build for other platforms.

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (zeiler) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:53]: oh removed this section as those vars aren't used.

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (zeiler) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:1]: yeah i'll leave that part to you after this one merges. That would be the second part of trying to get pip installs later. 

This does run already as is without copying over these libraries. I think the cuda stuff gets mounted in known places to torch automatically so we shouldn't have to copy lots of these.

File: clarifai/runners/dockerfile_template/Dockerfile.cuda.template
Comment (zeiler) [inline clarifai/runners/dockerfile_template/Dockerfile.cuda.template:18]: @wemoveon2 do we need this you think? I'm worried it'll instal some different version of python from what is in the base image.

Comment (wemoveon2) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:16]: its the header files needed to build Python extensions. i don't think it will affect installed ver

Comment (zeiler) [reply clarifai/runners/dockerfile_template/Dockerfile.cuda.template:16]: great

**Review by sainivedh — APPROVED** [review]
Just approving this to unblock.

**Pull request merged**
